jira dora (#2347)
port-labs/port-docs: docs/guides/all/setup-dora-metrics-jira.md

---
sidebar_position: 11
displayed_sidebar: null
---

import Tabs from "@theme/Tabs"
import TabItem from "@theme/TabItem"

# Set up deployments and incidents using Jira issues

The [Create & track DORA metrics in your portal](/guides/all/create-and-track-dora-metrics-in-your-portal) guide walks you through installing the DORA metrics experience in Port, using GitHub and PagerDuty as the default deployment and incident methods.

This guide assumes you have already completed that setup and shows how to adapt it for Jira users.

> By following this guide, you'll define deployments and incidents in Port using Jira issues. By the end, you'll leverage DORA metrics using Jira, providing accurate deployment and incident insights in your portal.

Reviewer: ```suggestion
By following this guide, you will see how to define deployments and incidents in Port using Jira issues. By the end, you will be able to leverage DORA metrics using Jira, providing accurate deployment and incident insights in your portal.
```

## Prerequisites

To set up Jira issues for DORA metrics in your portal, you will need:

- Admin permissions (to install the experience and execute self-service actions).
- An installed DORA Metrics experience, based on: [Create & track DORA metrics in your portal](/guides/all/create-and-track-dora-metrics-in-your-portal).
- An installed [Jira integration](/build-your-software-catalog/sync-data-to-catalog/project-management/jira/).

## Configure your deployments and incidents

> 1. **Map Jira issues as deployments or incidents**

Reviewer: make the steps headers instead of a numbered list. (use ###)

> - Navigate to **Builder** → **Data Sources** and select your Jira integration.

Reviewer: make the "Data sources" a link directly to the data sources page, and then you don't need to say builder-> ...

- Add new mappings to ingest Jira issues as deployment or incident entities.

:::info Technical note: mapping deployments and incidents

> These mappings ingest Jira events into the `dora_deployment_event` or `dora_incident_event` blueprints. For more details, visit [Create & track DORA metrics in your portal](/guides/all/create-and-track-dora-metrics-in-your-portal).

Reviewer: ```suggestion
These mappings ingest Jira events into the `dora_deployment_event` or `dora_incident_event` blueprints. For more details, see the [Create & track DORA metrics in your portal](/guides/all/create-and-track-dora-metrics-in-your-portal).
```

           lead_time_hours: >-
           if (.fields.resolutiondate != null and .fields.created != null)
           then ((.fields.resolutiondate[0:19] + "Z" | fromdateiso8601) -
           (.fields.created[0:19] + "Z" | fromdateiso8601)) / 86400 else null
           end
    ```

**Incidents using Jira issues:**

```yaml showLineNumbers
- kind: issue
  selector:
  query: 'true'
  jql: >-
      ((statusCategory != Done) OR (created >= -1w) OR (updated >= -1w))
  port:
  entity:
      mappings:
      identifier: .key
      title: .fields.summary
      blueprint: '"dora_incident_event"'
      properties:
          incident_type: '"Jira Issue"'
          description: .fields.description.content| .. | objects | select(.type? == "text") | .text
          incident_url: ( .self | split("/") | .[0:3] | join("/") ) + "/browse/" + .key
          created_at: .fields.created
          resolved_at: .fields.resolutiondate
          time_to_resolve: >-
          if (.fields.resolutiondate != null and .fields.created != null)
          then ((.fields.resolutiondate[0:19] + "Z" | fromdateiso8601) -
          (.fields.created[0:19] + "Z" | fromdateiso8601)) / 86400 else null
          end
    ```

> :::info Note
> You can map Jira custom fields to track additional incident information. Here's an example of mapping custom fields for urgency, status, and priority:
> :::

Reviewer: ```suggestion
You can map Jira custom fields to track additional incident information. Here's an example of mapping custom fields for urgency, status, and priority:
```

```yaml
urgency: .fields.customfield_00000
status: .fields.customfield_00000
priority: .fields.customfield_00000
```

> - Save the mappings. Port will begin ingesting matching Jira issues as deployments or incidents.

Reviewer: ```suggestion
After you save the mappings, Port will begin ingesting matching Jira issues as deployments or incidents.
```

> ### **(Optional) Filtering or tagging**
> 
>     - You can filter or tag deployments and incidents by project, issue types, statuses, or other criteria.
>     This can be accomplished by using JQL (Jira Query Language) within your mapping configuration. For additional information, refer to [JQL documentation]( https://docs.port.io/build-your-software-catalog/sync-data-to-catalog/project-management/jira/#jql-support-for-issues)

Reviewer: ```suggestion
### Filtering or tagging (optional)

You can filter or tag deployments and incidents by project, issue types, statuses, or other criteria.  
This can be accomplished by using JQL (Jira Query Language) within your mapping configuration. For additional information, refer to [JQL documentation]( https://docs.port.io/build-your-software-catalog/sync-data-to-catalog/project-management/jira/#jql-support-for-issues).
```
